torch/distributed/_shard/sharded_tensor/interface.py

```diff
@@ -44,15 +54,41 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
 
     @abc.abstractmethod
     def sharding_spec(self) -> shard_spec.ShardingSpec:
+        """
+        Returns the ShardingSpec for the subclass instance of ShardedTensorInterface.
+
+        Subclass of :class:`ShardedTensorInterface` could re-use existing :class:`ShardingSpec`
+        implementations like :class:`ChunkShardingSpec`, or implement its own custom spec.
+        """
         raise NotImplementedError()
 
     @abc.abstractmethod
     def local_shards(self) -> List[Shard]:
+        """
+        Returns a list of :class:`Shard' corresponding to the local shards for this rank.
+        Returns an empty list if the current rank does not host any shards.
+
+        Subclass of :class:`ShardedTensorInterface` should implement this method if it
+        choose to re-use :class:`Shard` for saving local shards.
+        """
         raise NotImplementedError()
 
     @abc.abstractmethod
     def reshard(self, resharding_spec: shard_spec.ShardingSpec) -> "ShardedTensorInterface":
+        """
+        Reshard a sharded tensor given the ``resharding_spec``.
+
+        Subclass of :class:`ShardedTensorInterface` should implement this method if it choose
+        to provide the resharding functionality.
+        """
         raise NotImplementedError()
 
     def __hash__(self):
+        """
+        By default implementation to define behavior when `hash()` is called on an instance
+        of class.
+
+        Subclass of :class:`ShardedTensorInterface` could override this method if it choose
+        to return a different result.
+        """
```

Comment:
Oh I guess we don't need docs for this method since it is not a public method rather a private magic method.